### PR TITLE
Updated the nullable type snippets

### PR DESCRIPTION
**Pull request by pkulikov**
Updated snippets for the dotnet/docs#6676

### CODE REVIEW

File: snippets/csharp/programming-guide/nullable-types/NullableTypesUsage.cs [hunk 1]
```diff
@@ -47,22 +51,23 @@ private static void AccessValueOfNullableType()
...
+        private static void ExplicitCast()
+        {
             // <Snippet5>
-            int? m;
-            int m2 = 10;
-            m = m2;  // Implicit conversion.
+            int? n = null;
+
+            //int m1 = n;    // Doesn't compile.
+            int n2 = (int)n; // Compiles, but throws an exception if n is null.
```
Comment (BillWagner) [inline snippets/csharp/programming-guide/nullable-types/NullableTypesUsage.cs]: Now that pattern matching is a thing, I'd really like to see a sample that does something like this:

```csharp
  public static void Main()
  {
    int? thing = 5;
    int? notAThing = default;

    TestValue(thing);
    TestValue(notAThing);
  }
  private static void TestValue(int? nullableThing)
  {
    if (nullableThing is int number)
    {
      Console.WriteLine(number);
    } else
    {
      Console.WriteLine("thing is default value");
    }

  }
```

**Comment by pkulikov** [comment]
@BillWagner thank you for suggestion! I've added the code example with pattern matching.

File: snippets/csharp/programming-guide/nullable-types/NullableTypesUsage.cs [hunk 2]
```diff
@@ -47,41 +52,57 @@ private static void AccessValueOfNullableType()
...
-            int n2 = (int)n;   // Compiles, but throws an exception if n is null.
-            int n3 = n.Value;  // Compiles, but throws an exception if n is null.
+            int? z = 42;
+            if (z is int zValue)
+            {
+                Console.WriteLine($"z is {zValue}");
+            }
+            else
+            {
+                Console.WriteLine("z is undefined");
```
Comment (BillWagner) [inline snippets/csharp/programming-guide/nullable-types/NullableTypesUsage.cs]: This is a great example, but I wouldn't use the word "undefined". Instead, I'd say "z does not have a value" or "z is null"

Comment (pkulikov) [reply snippets/csharp/programming-guide/nullable-types/NullableTypesUsage.cs]: @BillWagner I've updated phrasing, also in other examples.

**Comment by BillWagner** [comment]
Thanks again @pkulikov I've just :shipit: this, and I'll look at dotnet/docs#6676 now.

**Pull request merged**